File: 1288-remove-covered-intervals/1288-remove-covered-intervals.py
class Solution:
    def removeCoveredIntervals(self, intervals: List[List[int]]) -> int:
        if len(intervals) == 1:
            return 1

        intervals.sort(key=lambda x: (x[0], -x[1]))
        res = 1
        right = intervals[0][1]
        for start, end in intervals:
            if end <= right:
                continue
            else:
                res += 1
                right = end
        
        return res

# array, sorting
# time: O(nlog(n))
# space: O(1)

File: 1288-remove-covered-intervals/1288-remove-covered-intervals.go
func removeCoveredIntervals(intervals [][]int) int {
	if len(intervals) == 1 {
		return 1
	}

	sort.Slice(intervals, func(i, j int) bool {
		if intervals[i][0] == intervals[j][0] {
			return intervals[i][1] > intervals[j][1]
		}
		return intervals[i][0] < intervals[j][0]
	})

	idx := 1
    for idx < len(intervals) {
        if intervals[idx-1][1] >= intervals[idx][1] {
            intervals = append(intervals[:idx], intervals[idx+1:]...)
        } else {
            idx++
        }
    }

    return len(intervals)
}

// array, sorting
// time: O(n^2)
// space: O(1)